### ISSUE TITLE
Improve UX To Clarify That Error Counts Include Auto-fixes

### ISSUE THREAD

**Issue by drernie** [oxsecurity/megalinter#2291]
**Describe the bug**
When I run mega-linter on [my project](https://github.com/quiltdata/nf-quilt/actions/runs/4001321577/jobs/6867430257), it complains of one error:

❌ Linted [GROOVY] files with [npm-groovy-lint]: Found 1 error(s) - (22.15s) (expand for details)

However, I just fixed the last error (a weird NglParseError). In fact, the summary shows 0:
```
- Using [npm-groovy-lint v11.1.1] https://megalinter.io/6.18.0/descriptors/groovy_npm_groovy_lint
  Failure: 0 error(s) have been found 
[169](https://github.com/quiltdata/nf-quilt/actions/runs/4001321577/jobs/6867430257#step:4:183)
  Failure: 0 error(s) have been found 
```

**To Reproduce**

1. Go to https://github.com/quiltdata/nf-quilt/actions/runs/4001321577/jobs/6867430257
2. Click on the red ❌
3. Scroll down to `--Error detail`
4. See that both sections show `Failure: 0 error(s) have been found`

**Expected behavior**
If there are zero errors reported by the linter, then the summary should also have zero errors.

**Screenshots**

<img width="731" alt="Screenshot 2023-01-24 at 3 31 51 PM" src="https://user-images.githubusercontent.com/19791/214444268-3d383618-7666-4814-9265-3c482947e10f.png">

**Additional context**

The last few errors were exceeding strange NglParseErrors, of the form:
```
168 error Unexpected input: '=' @ line 168, column 67. NglParseError
```
I couldn't figure them out, but managed to make them go away by breaking up the affected lines.
Yet even with the the actual error count down to zero, the summary still shows "1" somewhere.

**Comment by drernie** [comment]
Things are now even more strange. I filtered it down to a single file that had one INFO error, which I fixed.
So now there aren't any listed issues, yet it somehow reports 3 issues, AND fails.

<img width="691" alt="Screenshot 2023-01-25 at 6 59 57 AM" src="https://user-images.githubusercontent.com/19791/214599934-53e1ea5f-def7-4096-8eeb-6694275ad93e.png">

**Comment by echoix** [comment]
Does the number of errors/the status come from the exit code? Maybe take a look at what is the exit code of the linter for investigating.

**Comment by drernie** [comment]
The plot thickens :-)

I replaced calling the Marketplace action with the example template, and now it runs properly and uploads the artifacts!

https://github.com/quiltdata/nf-quilt/actions/runs/4007539130/jobs/6880509055

However, it seems to run again (and fail) while trying to create the pull_request.  

<img width="475" alt="Screenshot 2023-01-25 at 7 56 02 AM" src="https://user-images.githubusercontent.com/19791/214611498-69d2c8d7-f49f-43a8-a48f-05f21e356051.png">

This is enough to get me moving forward, but it makes me wonder if there is a bug or mis-configuration somewhere, perhaps in the GitHub action. Will keep you posted if I find something more concrete.

**Comment by drernie** [comment]
> look at what is the exit code of the linter

Interesting. It seems to be calculated here:
https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/output.js

And set here:
https://github.com/nvuillam/npm-groovy-lint/blob/master/lib/groovy-lint.js    
// Exit with code 1 if failon, failonerror, failonwarning or failoninfo is set

Which I guess is reasonable, as long as it still saves the output.  That may be an issue with the GitHub Action, or my config, but we can close this issue. Thanks!

**Comment by nvuillam** [comment]
There is probably an issue counting the errors :/
But once there is no error, is the job status green ?

**Comment by drernie** [comment]
Sort of. If there's no errors, warnings, or info then it will turn green.  But sometimes it will show a "count" of 3 info messages, even though it does not list any actual such messages in the log.

**Comment by drernie** [comment]
Ah! It turns out that when it auto-fixes an error, it won't bother reporting it explicitly, but will still count it.
Once I applied all the auto-fixes, it turned green :-)

**Comment by Kurt-von-Laven** [comment]
Wow, nice discovery :). It seems like this is a UX issue then as we could probably communicate this more clearly.

**Comment by drernie** [comment]
Thanks so much! FYI, two possible affordances that would have helped me:
1. Log **all** the notices, including for auto-fixes
2. Log a 'summary' totaling up (just) the auto-fixes, so it is obvious why the total doesn't match the logged

I should perhaps mention that one reason this was so confusing for me is that I could NOT get the GitHub action to auto-generate a PR or commit containing the auto-fixes. In fact, when I used the 'naive' [marketplace action](https://github.com/marketplace/actions/megalinter) I didn't even get an artifact; 

```
- name: MegaLinter
  uses: oxsecurity/megalinter@v6.18.0
```

I had to paste in the the 'complete' action from the instructions, which DID create the artifact -- but I couldn't make it generate a PR or commit.

This was my first major use of GitHub Actions, so I assumed it was some silly error on my part, but I can open that as a separate issue if you like.
